### DISCUSSION TITLE
After Foundations: Lesson on starting early with social networking

### DISCUSSION THREAD

**Discussion by kasugaijin** [TheOdinProject/curriculum#26246] [General]
### Describe your suggestion

A few of us who help out in the getting hired channel have noticed a consistent stream of similar questions. Namely at what point in the curriculum do I start applying for jobs? How do I improve my chances/stand out from the crowd? How do I network?

So, we thought it would be useful to provide guidance on networking in a lesson. We consider a lesson soon after Foundations to be a practical place, because, taken early, this advice can pay off in the long run and is ideally not left to the last minute. The lesson content would provide some advice on how to improve chances of success by building a social network. A rough draft of the content can be found here: 

https://docs.google.com/document/d/1FHZp0gGEYkQM5w0JJqSEZ1n-82P8Poc7esaoWWGXkr8/edit?usp=sharing

Please let me know your thoughts on this.

### Path

~~Foundations~~
After foundations

### Lesson Url

see link to draft lesson in description

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

kasugaijin

**Comment by ht-l1** [comment]
read the doc you wrote and just want to say it looks really great and helpful!
hope the reviewers can add this new content in as part of the official curriculum :)

**Reply by kasugaijin** [reply]
Thank you. I have updated the document. Please take a read and let me know if you can think of ways to improve it.

**Comment by CouchofTomato** [comment]
I've converted this to a discussion because it needs a bit more thought.

**Reply by kasugaijin** [reply]
Ok. What's next from here?

**Reply by CouchofTomato** [reply]
Usually we'd look to see how much interest there is in this from any feedback on the discussion. Networking is absolutely beneficial, but it's a delicate balance on where to put it. Without a bit more data it's not easy to justify. Maybe share a link to this on discord and seek opinions from others?

**Reply by kasugaijin** [reply]
How about now :dancing_women:

**Reply by kasugaijin** [reply]
bump

**Comment by bycdiaz** [comment]
I think I like the idea of the content. But not sure if foundations feels good to me. In my head, this would feel better to me early in the next track.

**Reply by Beliver12** [reply]
I agree with adding it into next track. but didnt read it yet, will add more coments when i read it. but i want to start reading about networking for sure while im studying TOP.

**Reply by RolandStuder** [reply]
I agree with other comments that foundations is too early. 

I observe a lot of people asking about when to start applying when the reach the react section, or in the ruby path maybe after the forms project or so (less sure there). I think that would be a good moment to tell them: "Nah, you are probably not ready to apply to positions yet, but you can start networking, going to events…"

**Reply by kasugaijin** [reply]
Take a read now and let me know if there's anything missing/needs changing :)

**Comment by crespire** [comment]
I'm on the same line of comment as Carlos. I think it's good to put this idea into folks' heads somewhat early, but I think there's a risk that we don't necessarily prepare folks well enough at this stage of the curriculum to not have them spin their wheels.

As an example, I think open source contributions are a really easy way to *demonstrate* skills rather than just talking about them. But at foundations, I would guess that the majority of the learners will not be ready to contribute to projects at a high level and have that work add value in the context of a job search. I understand non-code contributions are important as well and I am not denigrating those kinds of contributions (those are how I got started!), but in the context of a job search, those types of contributions are pretty low value for demonstrating mastery.

Is there a way we could think about splitting this content up? I think some of it definitely belongs in foundations. In particular, treating TOP Discord as a community where you can make good connections for future growth, and also thinking about how one wants to "build their brand" being flagged early are excellent.

On the other hand, the open source contribution section, building a network on LinkedIn, and maybe writing content seem to me to be more appropriate for later in the curriculum. My sense is that by encouraging some of this so early, we might be sending some learners on wild goose chases that may end up undermining their credibility later, once they're further along in the curriculum.

**Reply by kasugaijin** [reply]
Thanks for the thoughtful feedback. The draft lesson is now just focusing on Networking. Please take a look and let me know if you think anything is missing, and where your gut would place this in the curriculum.

**Comment by Advaitmenon1106** [comment]
I believe the content will be a great addition, but I think (as said by few others) Foundations should just be reserved for learning, and by the time one gets to the Paths (and their knowledge about the course and how it develops a learning curve for the students, matures a bit more), maybe they should encounter this content as a part of a separate lesson later in the Paths.

As somebody who has completed Foundations fairly recently (June, this year), I've myself found myself to be very enthusiastic about my web dev skills after foundations till I saw the Javascript path and its lessons (lol). So I believe putting this early might also add on to a potential overestimation of one's knowledge/skills (I might be wrong here though), only for it to get deflated even more after seeing just how vast the learning process even is, in the paths.

**Reply by kasugaijin** [reply]
This is a great point! I have re-written to focus on networking only. Please take a look and let me know if you can see anything that needs changing. I also agree, after foundations is ideal...where exactly, though, is the tricky question.

**Reply by Advaitmenon1106** [reply]
Looks good. Regarding the positioning, I'm a student myself, so I may not be the right person to recommend this.

Having said that, my intuition tells me the content would probably be well-placed somewhere in Getting Hired, like many others said, but also at a location where you all feel that the learners are ready/near-ready to be hired for a FrontEnd Job.

**Reply by kasugaijin** [reply]
Yes that's where it would logically sit in terms of the curriculums different modules. The main argument against that, and placing it earlier on, is that networking is something you should start sooner than later, and a lot of good networking time has passed before you get to the Getting Hired section.

**Reply by Advaitmenon1106** [reply]
That is a fair point. Yeah I admittedly didn't think of the time taken for networking, or that aspect atleast, my bad.

**Comment by pixie-cheeks** [comment]
Contributing to open source projects was one of the main reasons I started doing TOP. I personally did not and do not feel confident in being involved in those projects yet, as I've only recently started with the JavaScript Path. I definitely love the concept of more people looking forward to help the community and getting involved in networking.

**Reply by kasugaijin** [reply]
I have re-focused this lesson to cover networking only. However, if you want to collab on an open source lesson, let me know.

**Comment by RolandStuder** [comment]
**Separating open source from networking**?

I think contributing to open source is just a completely other difficulty level to networking like going to events, getting to know devs. The entry barrier to making a PR is just so much higher than going to some tech events and talking to people, joining discussions on discords and so on…

**Reply by kasugaijin** [reply]
Done! Focuses only on networking now.

**Comment by debanjankoley** [comment]
I think this will be a great addition to the curriculum. This is something I have personally asked about in the getting-hired channel as well and to have all the resources in the same place is amazing. Networking is a long process and I (beginner) personally don't have any problems having this in the foundations course.

**Reply by kasugaijin** [reply]
Re-written it to focus only on networking now :)

**Comment by 0xYoyo** [comment]
This will serve as a very valuable addition to the curriculum. 
I do however agree with what @RolandStuder pointed out. They should be separated.
While networking in my opinion could be applied as early as end of Foundations. 
Full on open source contributions can come in at the "Intermediate Git" section on the JS/Ruby course.

*Note - Basic Forking & Pull requests can be covered in the "Git Basics" section on Foundations.

**Reply by kasugaijin** [reply]
Yes good point. I have re-written the lesson to focus only on networking. Take a look!

**Comment by kasugaijin** [comment]
Great feedback everyone. I hear your suggestions and agree with this being trimmed down to focus on networking and including it after foundations. I will 'refactor' the document ready for another review.

**Reply by crespire** [reply]
Thanks for getting the conversation started!

**Comment by kasugaijin** [comment]
Ok I have re-worked the document based on the feedback above and would love another review! If you have any specific comments, or ideas for where I solicit for more, please add a comment in the document. Also please comment on here with your view on whether this is close to what we are after. 

https://docs.google.com/document/d/1FHZp0gGEYkQM5w0JJqSEZ1n-82P8Poc7esaoWWGXkr8/edit?usp=sharing

Thank you again!

**Comment by Brodypen** [comment]
I agree with a lot of others that this is better in the Getting Hired part of the course, however I agree it does make sense to do a lot of this knowledge as soon as you're done with foundations. The more you get out there in the public the more you will be seen.

**Reply by kasugaijin** [reply]
Yes that is the tough part, where does it live exactly? I think it easily slots into the getting hired module based on the gist of the content, but the timing of it would not be ideal.

**Comment by benzelcurry** [comment]
Perhaps this idea is a bit half-baked, but I feel that this content belongs in some sort of optional, interstitial section titled something along the lines of "(Optional) Things to consider before proceeding..." that goes between Foundations and wherever the student will be going next.

In this idea, it would almost serve as a prequel to the Getting Hired course, providing those who are seeking work with options that they can look into while progressing through the rest of TOP. If this route is taken, I think it would be best to split your lesson into two or three smaller lessons so that it doesn't feel so lonely sitting in its own section.

**Reply by kasugaijin** [reply]
Yeah I agree. It belongs in getting hired in so far as the gist of the lesson, but it's too late to focus on it when you're ready to start applying.

**Comment by reyes-dev** [comment]
I read through the second draft, and I think it's a good lesson. I like that you explain why networking is helpful, it's hugely reassuring to be able to ask people already working as developers for advice, guidance, and even code related questions. It can clear up a lot of doubt or confusion about certain things. 

One thing I was a bit skeptical about was the emphasis on 80% of jobs being unposted and the hidden job market, after all 73.6% of statistics are made up. Personally, I'm still lost on how to land the first job, so it may well be true, but I've also been to a ton of local meetups and haven't had luck landing a job through a connection (met tons of great people though!). My point is, I hope people don't take away from the lesson that they shouldn't try to apply to job postings at all.

Also, regarding online networking, and the whole approach of getting in contact with engineers/recruiters/whoever who work with companies you're interested in, is something I would like to see expanded on. I've heard a lot about how "cold emails" to recruiters, or directly messaging people at companies, etc... can give you a leg up. But in my mind, that seems kind of hard to do, aren't these busy people with lives, and in general with little relation to me? You could spend a lot of time engaging with content, and trying to befriend people but it leading nowhere, vs. spamming job apps and perhaps landing an interview or two. I'm just playing devil's advocate here, I honestly don't know what the right approach is. This definitely seems like a hotly debated thing in the tech community and the job hunt, at times, seems like a total enigma in the current market.

Overall though, I love the idea of this lesson. Networking locally has really changed my perspective as a developer and has been a greatly motivating force for me. Keep up the good work!

**Reply by kasugaijin** [reply]
Thank you for the detailed response. I think you have some good points here. The main one being it should be clear that networking is not a job guarantee. I agree. It is really hard to tell just how impactful networking will be, but I would hazard a guess most do not regret it, and that it helped more than it hindered (my N=1 anecdote, I got both of my jobs from networking on LinkedIn).

I also agree that the 80% hidden jobs stat is just a stat. I tried to qualify that, regardless of the true number, it's a real thing. 

I think there's a balance for sure in conveying that networking is important, but it's not the only thing to focus on nor a panacea.
